woocommerce/woocommerce-gutenberg-products-block
# Add billing data context

This PR adds billing data context to persist billing data properties to state. This PR also fixes various context bugs I found whilst working on it.

- Adds BillingDataContext to hold address, email, and phone. Removed the useBillingData hook which is no longer needed.
- Renames ShippingMethodDataContext to ShippingDataContext for consistency
- Removes billing data from checkout and payment data contexts.
- Added CheckoutProcessor component placeholder; will be using this in a follow up
- Fixes PHONE field - this is NOT shipping address data
- Fixes Shipping Context state updater - it was missing a spread
- Fixes shipping address "apartment" usage which does not exist. Should be address 2.

### How to test the changes in this Pull Request:

Interact with the checkout form and make sure field changes persist.

@nerrad If we can get this reviewed I'll move on to processing.
